packages/toolpad-app/src/utils/recaptcha.ts

```diff
@@ -0,0 +1,31 @@
+export const getRecaptchaToken = async (siteKey: string): Promise<string> => {
+  await new Promise<void>((resolve) => grecaptcha.ready(resolve));
+  return grecaptcha.execute(siteKey, {
+    action: 'submit',
+  });
+};
+
+const SITE_VERIFY_URL = 'https://www.google.com/recaptcha/api/siteverify';
+const SCORE_THRESHOLD = 0.5;
+
+export const validateRecaptchaToken = async (
```

Comment:
@apedroferreira This file is mixing serverside-only code with client side code.   You might want to split that

Reply:
added the changes here https://github.com/mui/mui-toolpad/pull/1087/commits/8ec6c3dbff03c1e4201a1be183e12c0db71a87d1